user-ops-indexer/user-ops-indexer/src/indexer/v06/indexer.rs

```diff
@@ -0,0 +1,411 @@
+            .collect();
+
+        let user_ops: Vec<UserOp> = calldatas
+            .iter()
+            .zip(log_bundles.iter())
+            .enumerate()
+            .map(|(i, (calldata, log_bundle))| {
+                let calldata = IEntrypointV06Calls::decode(calldata)?;
+                let (bundler, raw_user_ops): (Address, Vec<RawUserOperation>) = match calldata {
+                    IEntrypointV06Calls::HandleAggregatedOps(cd) => (
+                        cd.beneficiary,
+                        cd.ops_per_aggregator
+                            .into_iter()
+                            .flat_map(|agg_ops| {
+                                agg_ops
+                                    .user_ops
+                                    .into_iter()
+                                    .map(move |op| RawUserOperation {
+                                        user_op: op,
+                                        aggregator: Some(agg_ops.aggregator),
+                                        aggregator_signature: Some(agg_ops.signature.clone()),
+                                    })
+                            })
+                            .collect(),
+                    ),
+                    IEntrypointV06Calls::HandleOps(cd) => (
+                        cd.beneficiary,
+                        cd.ops
+                            .into_iter()
+                            .map(|op| RawUserOperation {
+                                user_op: op,
+                                aggregator: None,
+                                aggregator_signature: None,
+                            })
+                            .collect(),
+                    ),
+                    _ => bail!("can't recognize calldata selector in {}", calldata),
+                };
+                if raw_user_ops.len() != log_bundle.len() {
+                    bail!(
+                        "number of user ops in calldata and logs don't match {} != {}",
+                        raw_user_ops.len(),
+                        log_bundle.len()
+                    )
+                }
+                Ok(raw_user_ops
+                    .iter()
+                    .zip(log_bundle.iter())
+                    .enumerate()
+                    .filter_map(|(j, (raw_user_op, logs))| {
+                        match build_user_op_model(
+                            bundler,
+                            i as u64,
+                            j as u64,
+                            raw_user_op,
+                            logs,
+                            &tx_deposits,
+                        ) {
+                            Ok(model) => Some(model),
+                            Err(err) => {
+                                let logs_start_index = logs
+                                    .get(0)
+                                    .map(|l| l.log_index)
+                                    .flatten()
```

Comment:
Nit: you can use `and_then` instead
```suggestion
                                    .and_then(|l| l.log_index)
```